aptos-move/framework/aptos-stdlib/sources/cryptography/keyless.move

```diff
@@ -0,0 +1,83 @@
+/// This module implements the Keyless authentication scheme.
+
+module aptos_std::keyless {
+    use aptos_std::bcs_stream::{Self, deserialize_u8};
+    use std::string::String;
+    friend aptos_std::federated_keyless;
+
+    #[test_only]
+    use std::string::{utf8};
+    #[test_only]
+    use std::bcs;
+
+
+    // Error codes
+    //
+
+    /// There are extra bytes in the input when deserializing a Keyless public key.
+    const E_INVALID_KEYLESS_PUBLIC_KEY_EXTRA_BYTES: u64 = 1;
+
+    //
+    // Constants
+    //
+
+    //
+    // Structs
+    //
+
+    /// An *unvalidated* any public key: not necessarily an elliptic curve point, just a sequence of 32 bytes
+    struct PublicKey has copy, drop, store {
+        iss: String,
+        idc: vector<u8>
+    }
+
+    //
+    // Functions
+    //
+
+    /// Parses the input bytes into a keyless public key.
+    public fun new_public_key_from_bytes(bytes: vector<u8>): PublicKey {
+        let stream = bcs_stream::new(bytes);
+        let key = deserialize_public_key(&mut stream);
+        assert!(bcs_stream::has_remaining(&mut stream) == false, std::error::invalid_argument(E_INVALID_KEYLESS_PUBLIC_KEY_EXTRA_BYTES));
+        key
+    }
+
+    public fun deserialize_public_key(stream: &mut bcs_stream::BCSStream): PublicKey {
+        let iss = bcs_stream::deserialize_string(stream);
```

Comment:
Can you rename the arg in bcs_stream::deserialize_* functions to self so we can use the more concise move 2 syntax here? Either in this PR or as a separate PR

Reply:
Will do in separate PR